src/races/races.service.ts

```diff
@@ -294,4 +296,534 @@ export class RacesService {
 
     return [...uniqueRaces.values()]
   }
+
+  private getRaceLevel(level: string) {
+    // this helper just simplifies level to city/state/county/federal.
+    // it is used in p2v.
+    // from this larger list:
+    // "level"
+    // "city"
+    // "county"
+    // "federal"
+    // "local"
+    // "regional"
+    // "state"
+    // "town"
+    // "township"
+    // "village"
+
+    try {
+      level = level.toLowerCase()
+      if (
+        level &&
+        level !== 'federal' &&
+        level !== 'state' &&
+        level !== 'county' &&
+        level !== 'city'
+      ) {
+        level = 'city'
+      }
+      return level
+    } catch (e) {
+      this.logger.log('error at get-race-level', e)
+      return level
+    }
```

Comment:
Not sure we need this `try-catch` right?  The only thing that could `throw` here is if `level` ends up not a `String` somehow.. and that shouldn't happen since we've declared it as such in the method args.

Even if we did want to protect against a runtime possibility there, we'd wrap just the `.toLowerCase()` here, not the entire conditional and `return`.